File: Planning-SPA/src/app/view/prof/models/prof.ts
import { Matiere } from './../../matiere/models/matiere';
import { Classe } from '../../classe/models/classe';
import { Cours } from '../../cours/models/cours';
export class Prof {
  id: number;
  matieres: Matiere[];
  classes: Classe[];
  cours: Cours[];
  constructor(
    public nom: string,
    public prenom: string,
    public adresse: string,
    public mail: string,
    public telephone: number,
  ) {
    this.id = 0;
  }



}

File: Planning.API/Models/Prof.cs
using System;
using System.Collections.Generic;

namespace Planning.API.Models
{
    public class Prof
    {
        public int Id { get; set;}
        public string Nom {get; set;}
        public string Prenom {get; set;}
        public string Adresse {get; set;}
        public string Mail {get; set;}
        public string Telephone {get; set;}

        public ICollection<ProfMatiere> ProfMatieres { get; set;}
        public ICollection<ProfClasse> ProfClasses { get; set;}
        public ICollection<Cours> Cours {get; set;}
        public ICollection<Indisponibilite> Indisponibiltes {get; set;}
    }
}